PythonRegression/util/threading_logic/pool_logic.py

```diff
@@ -0,0 +1,27 @@
+from multiprocessing.dummy import Pool
+import multiprocessing as mp
+import logging
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+def start_pool(function,iterations,args):
+    #Assigns the process worker pool to either the cpu count or the number of nodes for the call, whichever is lower
+    pool = Pool(processes=len(args) if len(args) <= mp.cpu_count() else mp.cpu_count())
```

Comment:
Why is the number of "processes" limited by the cpu count?
1. We are talking about networking I/O which is limited by bandwidth not cpu.
2. Regardless of 1, since you are using the `dummy` implementation your "processes" are actually green threads as the GIL doesn't allow real parallelism, so what's the point of the limit? (If we bring (1) back, I think you do get parallelism since you are not utilizing the cpu).

In short you can just remove the limit.